libs/logger/logger_spdlog.cpp

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "logger/logger_spdlog.hpp"
+
+#include <mutex>
+
+#include <spdlog/sinks/stdout_color_sinks.h>
+#include <spdlog/spdlog.h>
+
+static const std::string kDefaultPattern =
+    R"([%Y-%m-%d %H:%M:%S.%F] [%L] [%n]: %v)";
+
+namespace logger {
+
+  const LogPatterns kDefaultLogPatterns = ([] {
+    LogPatterns p;
+    p.setPattern(LogLevel::kTrace,
+                 R"([%Y-%m-%d %H:%M:%S.%F] [th:%t] [%5l] [%n]: %v)");
+    p.setPattern(LogLevel::kInfo, kDefaultPattern);
+    return p;
+  })();
+
+  void LogPatterns::setPattern(LogLevel level, std::string pattern) {
+    patterns_[level] = pattern;
+  }
+
+  std::string LogPatterns::getPattern(LogLevel level) const {
+    for (auto it = patterns_.rbegin(); it != patterns_.rend(); ++it) {
```

Comment:
Looks like a candidate for `std::find_if`.

Reply:
IMHO it complicates the code:
```c++
    auto it = std::find_if(patterns_.rbegin(),
                           patterns_.rend(),
                           [level](const auto &level_and_pattern) {
                             return level_and_pattern.first <= level;
                           });
    return it == patterns_.rend() ? kDefaultPattern : it->second;
```
Do you like this variant better?

Reply:
No 🙈